### PR TITLE
Markdown list continuation and indentation

### PR DESCRIPTION
**Pull request by ivan3bx**
## Summary

Adds smart bullet list handling for markdown files in the ContentBox editor:

- **Enter key**: Continues the list on the next line with the same indentation and marker (or incremented number/letter for ordered lists). Pressing Enter on an empty bullet line clears the bullet and exits list mode.
- **Tab key**: Indents a bullet line by inserting a tab at the start, preserving the cursor's relative position.
- **Shift+Tab key**: Dedents a bullet line by removing one leading tab (if present), preserving cursor position.

## Implementation

New file `list_continuation.go` provides pure parsing logic: `parseBulletPrefix()` parses unordered (- *) and ordered (1., a.) bullet patterns, and `nextBulletPrefix()` generates the next continuation prefix with proper increment handling.

ContentBox intercepts Enter/Tab/Shift-Tab keys in markdown files and delegates to new handlers: `handleListContinuation()`, `handleListIndent()`, and `handleListDedent()`.

All changes are isolated to markdown files; plain text files continue to use default TextArea behavior.

## Testing

Added 20+ table-driven tests covering all bullet types, indentation, cursor positioning, multiline documents, and edge cases. All existing tests pass.

### CODE REVIEW

File: content_box.go [hunk 1]
```diff
@@ -209,6 +230,117 @@ func (b *ContentBox) InputHandler() func(event *tcell.EventKey, setFocus func(p
...
+	// Find the start of the current line.
+	lineStart := strings.LastIndex(text[:start], "\n")
+	if lineStart < 0 {
+		lineStart = 0
+	} else {
+		lineStart++ // skip past the newline
+	}
+
+	currentLine := text[lineStart:start]
+	indent, marker, rest, ok := parseBulletPrefix(currentLine)
```
[the remark this answers is not in the document]
Comment (ivan3bx) [reply content_box.go]: Good catch — fixed. Now uses the full line (lineStart..lineEnd) for bullet detection, while still inserting the newline at the cursor position. Added a test case for cursor positioned right after the marker.

File: content_box.go [hunk 2]
```diff
@@ -209,6 +230,117 @@ func (b *ContentBox) InputHandler() func(event *tcell.EventKey, setFocus func(p
 	})
 }
 
+// handleListContinuation checks the current line for a bullet prefix and
+// either continues the list or clears an empty bullet. Returns true if the
+// Enter key was consumed and should not be passed to TextArea.
+func (b *ContentBox) handleListContinuation() bool {
+	text := b.GetText()
+	_, start, _ := b.GetSelection()
+
```
[the remark this answers is not in the document]
Comment (ivan3bx) [reply content_box.go:239]: I don't think this warrants a fix. Having an active selection while pressing Enter on a bullet line is an extreme edge case in a TUI editor (requires shift-arrow to select, then Enter). The current behavior — inserting at the selection start — is reasonable. Other markdown editors typically just replace the selection with a newline and lose list context entirely. Adding selection normalization here adds complexity without meaningful benefit.

File: content_box.go [hunk 3]
```diff
@@ -209,6 +230,117 @@ func (b *ContentBox) InputHandler() func(event *tcell.EventKey, setFocus func(p
...
+		if _, _, _, ok := parseBulletPrefix(fullLine); !ok {
+			return false
+		}
+	}
+
+	// Only dedent if the line starts with a tab.
+	if lineStart < len(text) && text[lineStart] == '\t' {
+		b.Replace(lineStart, lineStart+1, "")
+		// Restore cursor to its original relative position (shifted back by the removed tab).
+		b.Replace(cursorPos-1, cursorPos-1, "")
```
[the remark this answers is not in the document]
Comment (ivan3bx) [reply content_box.go]: Good catch — fixed. Now clamps the new cursor position to lineStart to prevent underflow when the cursor is at the start of the line. Added a test case for this edge case.
